flink-core/src/main/java/org/apache/flink/configuration/Configuration.java

```diff
@@ -958,6 +959,15 @@ private String convertToString(Object o) {
 		} else if (o.getClass() == Duration.class) {
 			Duration duration = (Duration) o;
 			return String.format("%d ns", duration.toNanos());
+		} else if (o instanceof List) {
+			return ((List<?>) o).stream()
+				.map(e -> escapeWithSingleQuote(convertToString(e), ";"))
+				.collect(Collectors.joining(";"));
+		} else if (o instanceof Map) {
+			return ((Map<?, ?>) o).entrySet().stream()
+				.map(e -> escapeWithSingleQuote(escapeWithSingleQuote(e.getKey().toString(), ":") + ":"
+						+ escapeWithSingleQuote(e.getValue().toString(), ":"), ","))
```

Comment:
How about we make it a bit more readable?

```
				.map(e -> {
					String escapedKey = escapeWithSingleQuote(e.getKey().toString(), ":");
					String escapedValue = escapeWithSingleQuote(e.getValue().toString(), ":");

					return escapeWithSingleQuote(escapedKey + ":" + escapedValued, ",");
				})
```

Reply:
Make sense. I will update it.